flink-runtime/src/test/java/org/apache/flink/runtime/minicluster/MiniClusterITCase.java

```diff
@@ -686,6 +690,38 @@ public void initializeOnMaster(ClassLoader loader) {
         }
     }
 
+    @Test
+    public void testDefaultSizeOfIOPool() throws Exception {
+
+        final int numOfTMs = 3;
+        final int slotsPerTM = 7;
+
+        final MiniClusterConfiguration cfg =
+                new MiniClusterConfiguration.Builder()
+                        .setNumTaskManagers(numOfTMs)
+                        .setNumSlotsPerTaskManager(slotsPerTM)
+                        .setConfiguration(getDefaultConfiguration())
+                        .build();
+
+        try (final MiniCluster miniCluster = new MiniCluster(cfg)) {
+            miniCluster.start();
+            miniCluster.getDispatcherGatewayFuture().get();
+
+            // check io pool size.
+            List<TaskExecutor> taskManagers = miniCluster.getTaskManagers();
+            for (TaskExecutor taskExecutor : taskManagers) {
+                ThreadPoolExecutor ioExecutor = (ThreadPoolExecutor) taskExecutor.getIOExecutor();
+                assertThat(
+                        ioExecutor.getCorePoolSize(),
+                        is(MiniClusterConfiguration.DEFAULT_IO_POOL_SIZE));
+            }
+
+            assertThat(
+                    ((ThreadPoolExecutor) miniCluster.getIOExecutor()).getCorePoolSize(),
+                    is(MiniClusterConfiguration.DEFAULT_IO_POOL_SIZE));
+        }
+    }
```

Comment:
I think this test leaks a bit of implementation details (e.g. `ioExecutor` being a `ThreadPoolExecutor`). Hence, I would argue that it is ok to remove this test.

Reply:
The test does need to leak implementation details,  I think it is reasonable to delete it.